app/controllers/cypress/factories_controller.rb

```diff
@@ -46,5 +73,24 @@ def params_to_attributes(params)
 
         return attributes, should_validate
       end
+
+      def transform_hash(value)
+        value.transform_keys(&:to_sym).transform_values do |v|
+          if v.is_a?(Hash) && v.keys.all? { |k| k.match?(/^\d+$/) }
+            # Convert nested arrays to arrays of strings
+            v.values.map(&:to_s)
+          else
+            v
+          end
+        end
+      end
+
+      def create_or_build_factory(attributes, should_validate)
```

Comment:
Same as above: I think it is not actually a factory that is created or built here. So the method name might be confusing.